In [1]:
import numpy as np
import scipy.stats as sps

data = np.load('9_1.npy')
n = data.size
n

100

Воспользуемся построенным в теоретической задаче 9.2 равномерно наиболее мощным критерием проверки гипотезы $H_0: \theta = \theta_0$ против альтернатив $\theta > \theta_0$  и  $\theta < \theta_0$. <br>

а) $H_1: \theta > \theta_0$ <br>
Критерий: $S_1 = $ { $\sum X_i \leqslant Z_\alpha$ } <br>

б) $H_2: \theta < \theta_0$ <br>
Критерий: $S_2 = $ { $\sum X_i \geqslant Z_{1-\alpha}$ }, <br><br>
где $Z_\alpha \space и \space Z_{1-\alpha}$ - квантили гамма распределения
<br><br>

Для каждого $\theta_i, \space i=1,2,3$ воспользуемся критерием и найдём такое $\alpha$, начиная с которого 2 из гипотез $H_i: \theta = \theta_i$ будут отвергаться, а третья - нет. Тогда значение $\theta_i$, не отвергнутое критериями, и будет истинным значением параметра $\theta$.

In [2]:
Statistik = sum(data)
Statistik

97.307961386590719

In [3]:
theta_1 = 0.9
theta_2 = 1
theta_3 = 1.1

In [4]:
def est(theta):
    for alpha in np.arange(0.3, 0.5, 0.01):
        ok = False
        quantile_alpha = sps.gamma.ppf(alpha, n, theta)
        quantile_one_minus_alpha = sps.gamma.ppf(1 - alpha, n, theta)

        if (Statistik > quantile_alpha) and (Statistik < quantile_one_minus_alpha):
            ok = True

        print(alpha, ok)

In [5]:
est(theta_1)

0.3 True
0.31 True
0.32 True
0.33 True
0.34 True
0.35 True
0.36 True
0.37 True
0.38 False
0.39 False
0.4 False
0.41 False
0.42 False
0.43 False
0.44 False
0.45 False
0.46 False
0.47 False
0.48 False
0.49 False


In [6]:
est(theta_2)

0.3 True
0.31 True
0.32 True
0.33 True
0.34 True
0.35 True
0.36 True
0.37 False
0.38 False
0.39 False
0.4 False
0.41 False
0.42 False
0.43 False
0.44 False
0.45 False
0.46 False
0.47 False
0.48 False
0.49 False


In [7]:
est(theta_3)

0.3 True
0.31 True
0.32 True
0.33 True
0.34 True
0.35 True
0.36 True
0.37 False
0.38 False
0.39 False
0.4 False
0.41 False
0.42 False
0.43 False
0.44 False
0.45 False
0.46 False
0.47 False
0.48 False
0.49 False


# Вывод:

Видно, что при $\alpha$, равном 0.37, $\theta_1 = 0.9$ не отвергается, в отличие от других. <br>
Значит, истинное значение параметра $\theta$ равно 0.9.